src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -1227,7 +1230,52 @@ public Response getMigrationInBulkUpload(
 		try {
 			List<Migration> migrations = repository
 					.getMigrationsInBulkUpload(bulk_upload_id);
-			return Response.status(Response.Status.OK).entity(migrations)
+			
+			HashMap<String, Object> statusMap = new HashMap<String, Object>();
+			List<String> sitesList = new ArrayList<String>();
+			HashMap<String, Object> totalMap = new HashMap<String, Object>();
+			int errorSiteCount = 0;
+			for (Migration m : migrations )
+			{
+				String siteId = m.getSite_id();
+				String siteStatus = Utils.STATUS_SUCCESS;
+				String siteStatusString = m.getStatus();
+				JSONObject siteStatusJson = new JSONObject(siteStatusString);
+				// look the tools attribute and find resource tool
+				JSONArray itemizedJSONArray = (JSONArray) siteStatusJson.get("data");
+				for (int iItem = 0; itemizedJSONArray != null && iItem < itemizedJSONArray.length(); ++iItem) {
+					JSONObject itemJSON = itemizedJSONArray.getJSONObject(iItem);
+					String path = itemJSON.getString("path");
+					String status = itemJSON.getString("status");
+					if (!path.endsWith("/") && status.indexOf("Box upload successful for file") == -1)
+					{
+						// file path did not end with "/"
+						// and if there is error, status message won't have String "Box upload successful for file"
+						// set the site migration status to be failure
+						siteStatus = Utils.STATUS_FAILURE;
+						errorSiteCount++;
+						break;
+					}
+				}
+				sitesList.add(siteId + ": " + siteStatus);
```

Comment:
Since the export expects a JSON array object, it is simpler to have the List as original object type. It would be much more work if we start with a Map object.